app/Http/Controllers/Api/Device/DeviceController.php

```diff
@@ -0,0 +1,336 @@
+     *             "device_id": 1,
+     *             "hostname": "localhost",
+     *             "sysName": "example.librenms.org",
+     *             "ip": null,
+     *             "community": "public",
+     *             "authlevel": "noAuthNoPriv",
+     *             "authname": "",
+     *             "authpass": "",
+     *             "authalgo": "MD5",
+     *             "cryptopass": "",
+     *             "cryptoalgo": "AES",
+     *             "snmpver": "v2c",
+     *             "port": "161",
+     *             "transport": "udp",
+     *             "timeout": null,
+     *             "retries": null,
+     *             "snmp_disable": "0",
+     *             "bgpLocalAs": null,
+     *             "sysObjectID": ".1.3.6.1.4.1.8072.3.2.10",
+     *             "sysDescr": "Linux 3.10.0-693.17.1.el7.x86_64 #1 SMP Thu Jan 25 20:13:58 UTC 2018 x86_64",
+     *             "sysContact": "Root <root@localhost> (configure /etc/snmp/snmp.local.conf)",
+     *             "version": "3.10.0-693.17.1.el7.x86_64",
+     *             "hardware": "Generic x86 64-bit",
+     *             "features": null,
+     *             "location": "NOC",
+     *             "os": "linux",
+     *             "status": "1",
+     *             "status_reason": "",
+     *             "ignore": "0",
+     *             "disabled": "0",
+     *             "uptime": "3805884",
+     *             "agent_uptime": "0",
+     *             "last_polled": "2018-05-18 09:25:30",
+     *             "last_poll_attempted": null,
+     *             "last_polled_timetaken": "22.39",
+     *             "last_discovered_timetaken": "10.63",
+     *             "last_discovered": "2018-05-18 06:34:41",
+     *             "last_ping": "2018-05-18 09:25:30",
+     *             "last_ping_timetaken": "0.35",
+     *             "purpose": null,
+     *             "type": "server",
+     *             "serial": null,
+     *             "icon": "http://example.org/images/os/linux.svg",
+     *             "poller_group": "0",
+     *             "override_sysLocation": "0",
+     *             "notes": null,
+     *             "port_association_mode": "1"
+     *         }
+     *     }
+     */
+    public function store(Request $request)
+    {
+        $this->validate($request, [
+            'hostname' => 'required',
+            'version'   => [
+                \Illuminate\Validation\Rule::in(['v1', 'v2c', 'v3'])
+            ],
+            'v3.authlevel' => 'required_if:version,v3',
+            'v3.authname' => 'required_if:version,v3',
+            'v3.authpass' => 'required_if:version,v3',
+            'v3.authalgo' => 'required_if:version,v3',
+            'v3.cryptopass' => 'required_if:version,v3',
+            'v3.cryptoalgo' => 'required_if:version,v3',
+        ]);
+        $snmpver = 'v2c';
+        $hostname     = $request->hostname;
+        $port         = (isset($request->port) ? $request->port : Config::get('snmp.port'));
```

Comment:
@pheinrichs FYI, you can simplify to this `$request->get('port', Config::get('snmp.port', 161))` (and add the default as 161) Same goes for anything else, were you need to supply a default, like snmpver

Reply:
Oh perfect, thanks!